### PR TITLE
Move performance testing driver scripts to `scripts/performance`

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This moved the performance testing driver scripts from `pyomo-model-libraries` into `scripts/performance` to make it easier for derived projects to reuse the functionality.

## Changes proposed in this PR:
- Copy `main.py` and `compare.py` scripts from https://github.com/Pyomo/pyomo-model-libraries
- Update timing handler to make it easier for derived projects to hook into it
- remove old performance drivers that have been superseded by the current ones in pyomo-model-libraries 

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: scripts/performance/main.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    recorder = DataRecorder(results[1])
+    unittest.pytest.main(argv, plugins=[recorder])
+    gc.collect()
+    gc.collect()
+    return results
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(
+        epilog="Remaining arguments are passed to nosetests"
```
Comment (blnicho) [inline scripts/performance/main.py:149]: Is "nosetests" still correct here?

**Review by blnicho — APPROVED** [review]

File: scripts/performance/main.py (hunk 1 above)
Comment (mmundt) [reply scripts/performance/main.py:149]: Oh definitely not. Not correct at all.

Comment (mrmundt) [reply scripts/performance/main.py:149]: It is definitely not right. Should be pytest.

File: scripts/performance/compare.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (jsiirola) [reply scripts/performance/compare.py:1]: Nope.  Needed to make the script executable.

File: scripts/performance/main.py (hunk 1 above)
Comment (jsiirola) [reply scripts/performance/main.py]: fixed.

**Review by mrmundt — APPROVED** [review]

**Comment by mrmundt** [comment]
Besides that tests are now failing, I'm cool with all of this.

**Pull request merged**